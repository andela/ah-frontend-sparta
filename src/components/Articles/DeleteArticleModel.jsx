import React from 'react';
import './article.scss';

const deleteArticleModal = ({ title, slug, deleteAnArticle }) => (
  <div
    className="modal fade"
    id="deleteArticleModal"
    tabIndex="-1"
    role="dialog"
    aria-labelledby="deleteArticleModal"
    aria-hidden="true"
  >
    <div
      className="modal-dialog custom-model-dialog"
      role="document"
    >
      <div className="modal-content">
        <div className="modal-header">
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
          Delete Article
          </h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          Are you sure , you want to delete ?
          <br />
          <b className="delete-article-title">
            Title :
            {' '}
            {title}
          </b>
        </div>
        <div className="modal-footer">
          <button
            type="button"
            className="btn btn-success"
            data-dismiss="modal"
            id={`cancelModal${slug}`}
          >
          Cancel

          </button>
          <button
            type="button"
            id="DeleteArticleBtn"
            className="btn btn-danger"
            onClick={() => {
              deleteAnArticle(slug);
            }}
          >
          Ok
          </button>
        </div>
      </div>
    </div>
  </div>
);

export default deleteArticleModal;
