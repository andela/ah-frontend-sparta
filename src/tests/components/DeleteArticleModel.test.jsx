import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import DeleteArticleModel from '../../components/Articles/DeleteArticleModel';

describe('Delete Article  Component', () => {
  let mountApp;
  const props = {
    title: '',
    slug: '',
    deleteAnArticle: jest.fn(),
  };
  beforeEach(() => {
    mountApp = renderer.create(
      <Router>
        <DeleteArticleModel {...props} />
      </Router>,
    );
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });

  it('should call a delete an article function on click', () => {
    const wrapper = mount(
      <DeleteArticleModel {...props} />,
    );
    const okButton = wrapper.find('#DeleteArticleBtn').first();
    okButton.simulate('click');
    expect(props.deleteAnArticle).toHaveBeenCalled();
  });
});
