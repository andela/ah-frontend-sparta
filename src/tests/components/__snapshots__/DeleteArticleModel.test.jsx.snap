// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Article  Component Should render without crashing 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="deleteArticleModal"
  className="modal fade"
  id="deleteArticleModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog custom-model-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel"
        >
          Delete Article
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        Are you sure , you want to delete ?
        <br />
        <b
          className="delete-article-title"
        >
          Title :
           
          
        </b>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-success"
          data-dismiss="modal"
          id="cancelModal"
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          id="DeleteArticleBtn"
          onClick={[Function]}
          type="button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</div>
`;
